cpp/oneapi/dal/table/backend/interop/sycl_table_adapter_dpc.cpp

```diff
@@ -0,0 +1,289 @@
+/*******************************************************************************
```

Comment:
Why header is named `sycl_table_adapter` but source file is `sycl_table_adapter_dpc`?

Reply:
It needs to be compiled only for DPC++. It's our convention to put `_dpc` at the end.